### DISCUSSION TITLE
way to setup without Docker?

### DISCUSSION THREAD

**Discussion by socphoenix** [gramps-project/gramps-web#226] [General]
I'd like to try to port this to FreeBSD so it could be run in a jail. They don't have docker as a system option. Not sure if anyone could give some advice on where to start to try to decouple this from docker.

**Comment by DavidMStraub** [comment]
Hi, are you after a development or production setup?

**Reply by socphoenix** [reply]
Goal would be production. Other web apps for FreeBSD that have been ported basically dump the web interface into a folder in /usr/local/www/"package_name" and you point the web server config there. The ports system lets me list any dependencies to install at the same time, I'm just struggling to figure out exactly what the docker compose command is setting up as far as directories and required packages for this and the backend.

**Comment by DavidMStraub** [comment]
Ok, just asking because the answer would be quite different in the two cases.

To deploy without docker, basically you can look at the dockerfiles: 

https://github.com/DavidMStraub/gramps-docker/blob/master/Dockerfile installs Gramps
https://github.com/gramps-project/gramps-webapi/blob/master/Dockerfile installs Gramps Web API and runs it with Gunicorn
https://github.com/gramps-project/Gramps.js/blob/main/Dockerfile simply copies the static compiled frontend to the "static" folder

To get the compiled frontend, you can just download `grampsjs-vXXX.tar.gz` from the [releases page](https://github.com/gramps-project/Gramps.js/releases).

**Reply by socphoenix** [reply]
Cool. Thank you so much for the pointers!
